### PR TITLE
intake process and prioritization

### PR DESCRIPTION
**Pull request by ultrasaurus**
Moving [doc](https://docs.google.com/document/d/1RUzTBHFqpEL27RvSRMJVcbQtqDj1v78eOu-5xEvVw6I/edit#)  to pull request.  Reviewed by TOC Liaissons @lizrice and @jbeda and co-chairs

Additional feedback welcome!

### CODE REVIEW

File: assessments/intake-process.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+SIG-Security has a volunteer team of subject matter experts and industry professionals dedicated to helping SIG-Security members, the TOC, and the larger CNCF community maintain an understanding of the current state of security in the cloud native ecosystem and helping cloud native projects succeed. 
+
+The following process describes how projects are prioritized for security assessments.
+
+# Authority
+
+* The [Security Assessment Facilitator](../governance/roles.md#security-assessment-facilitator) maintains the Assessment queue
+* A named chair provides oversight for the Security Assessment initiative. 
+This chair can adjust queue, and is also responsible for conveying TOC 
+instructions, when needed. (Sarah Allen is the current named Chair for Security Assessments.)
```
Comment (TheFoxAtWork) [inline assessments/intake-process.md]: Should we call out @ultrasaurus  in the doc as the current chair?  should the roles held by members go somewhere else?  want to ensure we update role changes one place instead of several places

File: assessments/intake-process.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Security Assessment Priorities & Pipeline Intake Process
+
+SIG-Security has a volunteer team of subject matter experts and industry professionals dedicated to helping SIG-Security members, the TOC, and the larger CNCF community maintain an understanding of the current state of security in the cloud native ecosystem and helping cloud native projects succeed. 
+
```
Comment (TheFoxAtWork) [inline assessments/intake-process.md]: Some formatting inconsistencies - line breaks versus line wraps.  Refer to [contributing.md](https://github.com/cncf/sig-security/blob/master/CONTRIBUTING.md)

File: assessments/intake-process.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+* The [Security Assessment Facilitator](../governance/roles.md#security-assessment-facilitator) maintains the Assessment queue
+* A named chair provides oversight for the Security Assessment initiative. 
+This chair can adjust queue, and is also responsible for conveying TOC 
+instructions, when needed. (Sarah Allen is the current named Chair for Security Assessments.)
+
+# Pre-conditions
+
+* The project is either a CNCF project OR an assertion that the project is cloud native (any objection must be resolved before an assessment would be considered)
+* The project has identified a project lead and has a written self-assessment
```
Comment (TheFoxAtWork) [inline assessments/intake-process.md]: Recommend re-writing this as it doesn't cover potential incubation projects (implied maybe?)  something maybe to the effect of:
Projects must meet the following preconditions in order to be placed in queue for security assessment:
* existing CNCF project
* eligible for CNCF incubation and intention to join (link to CNCF eligibility)

**Comment by lumjjb** [comment]
To address https://github.com/cncf/sig-security/issues/281

File: assessments/intake-process.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+CNCF community maintain an understanding of the current state of security in the
+cloud native ecosystem and helping cloud native projects succeed.
+
+The following process describes how projects are prioritized for security
+assessments.
+
+# Authority
+
+* The [Security Assessment
+  Facilitator](../governance/roles.md#security-assessment-facilitator) maintains
```
Comment (lumjjb) [inline assessments/intake-process.md]: I think this should be a direct link instead of a relative one?

File: assessments/intake-process.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+# Pre-conditions
+
+* The project is either a CNCF project OR an assertion that the project is cloud
+  native (any objection must be resolved before an assessment would be
+  considered)
+* The project has identified a project lead and has a written self-assessment
+
+# Intake priorities
+
+1. TOC requests SIG-Security review a specific project or adjust priorities a.
```
Comment (lumjjb) [inline assessments/intake-process.md]: perhaps make this as a sub bullet

1. abc
  a. def
  b. ghi
2. aaa

File: assessments/intake-process.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+   TOC request will not interrupt an ongoing assessment b. TOC requests may jump
+   the prioritized queue of projects waiting for an assessment
+2. Projects that have received a CNCF Security Audit will be reviewed within a
+   year of audit.
+3. CNCF Projects that request a review (or invited by SIG members). Prioritized
+   by project majority (Graduated projects will be highest priority, then
+   incubated projects, then sandbox.)
+4. Non-CNCF Projects that request a review (or invited by SIG members)
+
+## Updates and renewal
```
Comment (lumjjb) [inline assessments/intake-process.md]: Should this section be just 1 '#'? Looks unrelated to intake priority.

File: assessments/intake-process.md (hunk 1 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: addressed by moving who is filling what role to new section in the repo root README (and linking from here)

File: assessments/intake-process.md (hunk 3 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: I think "an assertion that the project is cloud native (any objection must be resolved before an assessment would be considered)" lightly covers projects that have submitted proposals.  

I am intentionally allowing for us to review a project that does not plan to join the CNCF (but is cloud native).  I don't think we'll do this often (maybe never), but it could come up if a project is widely used by CNCF projects and is already part of some other foundation and it would benefit CNCF projects or end-users to have an assessment of that project.

File: assessments/intake-process.md (hunk 4 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: not sure what you mean here

File: assessments/intake-process.md (hunk 5 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: done!

File: assessments/intake-process.md (hunk 6 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: done

File: assessments/intake-process.md (hunk 4 above)
Comment (lumjjb) [reply assessments/intake-process.md]: I was thinking of doing `/governance/roles.md#security-assessment-facilitator` instead of `../`

File: assessments/intake-process.md (hunk 3 above)
Comment (TheFoxAtWork) [reply assessments/intake-process.md]: LGTM

File: assessments/intake-process.md (hunk 1 above)
Comment (TheFoxAtWork) [reply assessments/intake-process.md]: LGTM

**Comment by lumjjb** [comment]
Is the intake priority queue going to live somewhere in the repo as well? Or is it something that is going to be managed through an issue (or github project)?

**Comment by ultrasaurus** [comment]
Currently the "priority queue" is a [matrix](https://github.com/cncf/sig-security/blob/master/assessments/assessment-matrix.md) and there's some discussing about refactoring that https://github.com/cncf/sig-security/issues/309 and https://github.com/cncf/sig-security/issues/206 -- I guess I could link to the matrix and we can just remember to fix the link if it gets renamed.

Will wait for @JustinCappos review of this PR and see what he thinks

File: assessments/intake-process.md (hunk 4 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: makes sense, will search for `..`

File: assessments/intake-process.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+The following process describes how projects are prioritized for security
+assessments.
+
+# Authority
+
+* The [Security Assessment
+  Facilitator](../governance/roles.md#security-assessment-facilitator) maintains
+  the Assessment queue
+* A named chair provides oversight for the Security Assessment initiative. This
+  chair can adjust queue, and is also responsible for conveying TOC
```
[the remark this answers is not in the document]
Comment (ultrasaurus) [reply assessments/intake-process.md]: Trying to be a little more explicit that chair can/must intervene if needed, but generally won't do much, unless asked (or if chair learns about an opportunity to highlight the work of the team)

> A named chair provides oversight for the Security Assessment initiative,
  responsible for liaising with the TOC: aligning prioritization with TOC needs 
  and goals by finding opportunities to highlight work of Security Assessment 
  team, resolving questions/concerns about prioritization, and serving as an
  escalation point for projects or SIG members, if needed.

File: assessments/intake-process.md (hunk 4 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: done

File: assessments/intake-process.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  instructions, when needed. (Sarah Allen is the current named Chair for
```
[the remark this answers is not in the document]
Comment (ultrasaurus) [reply assessments/intake-process.md]: reworded this and added section at the bottom, please review "Managing the assessment queue"

File: assessments/intake-process.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+2. Projects that have received a CNCF Security Audit will be reviewed within a
+   year of audit.
+3. CNCF Projects that request a review (or invited by SIG members). Prioritized
+   by project majority (Graduated projects will be highest priority, then
+   incubated projects, then sandbox.)
+4. Non-CNCF Projects that request a review (or invited by SIG members)
+
+## Updates and renewal
+
+The Security Assessment team will review assessed projects annually, focusing
```
[the remark this answers is not in the document]
Comment (ultrasaurus) [reply assessments/intake-process.md]: changed "will" to "aims to" so it's not a strict requirement, we'll work on expanding capacity to allow for this, but if we don't have bandwidth, we'll prioritize as makes sense

**Comment by ultrasaurus** [comment]
There are a couple of comments that I still need to review, but most of the feedback has been addressed.  Would welcome any new set of eyes to take a look!

File: assessments/intake-process.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  native (any objection must be resolved before an assessment would be
+  considered)
+* The project has identified a project lead and has a written self-assessment
+
+# Intake priorities
+
+1. TOC requests SIG-Security review a specific project or adjust priorities a.
+   TOC request will not interrupt an ongoing assessment b. TOC requests may jump
+   the prioritized queue of projects waiting for an assessment
+2. Projects that have received a CNCF Security Audit will be reviewed within a
```
[the remark this answers is not in the document]
Comment (ultrasaurus) [reply assessments/intake-process.md]: discussed in Dec group meeting... this is guidance

File: assessments/intake-process.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+The following process describes how projects are prioritized for security
+assessments.
+
+# Authority
+
+Team members are welcome to submit PRs to streamline process when priorities
+are clear based on the criteria below. As needed, specific leaders (identified
+in the [repo root README](/README.md@#security-assessments) will coordinate
+the decision-making process.
```
Comment (dshaw) [inline assessments/intake-process.md]: Add closing paren.

File: assessments/README.md [hunk 12]
```diff
@@ -2,39 +2,73 @@
...
-The primary goal is to reduce the risk from malicious attacks and accidental breaches of privacy. This process supports that goal in two ways:
+The primary goal is to reduce the risk from malicious attacks and accidental
+breaches of privacy. This process supports that goal in two ways:
 
-   * Clear and consistent process for communication increases detection & reduces time to resolve known or suspected vulnerability issues
-   * A collaborative evaluation process increases domain expertise within each participating project.
+   * Clear and consistent process for communication increases detection &
+     reduces time to resolve known or suspected vulnerability issues
+   * A collaborative evaluation process increases domain expertise within each
+     participating project.
```
[the remark this answers is not in the document]
Comment (ultrasaurus) [reply assessments/README.md]: I think this is addressed in #2 -- if you feel strongly about the addition, pls consider a follow-up PR!

File: assessments/README.md [hunk 13]
```diff
@@ -2,39 +2,73 @@
...
+   dependencies or complementary software
+4. next steps toward increasing security of the project itself and/or increasing
+   the applications of the project toward a more secure cloud native ecosystem
 
-Due to the nature and timeframe for the analysis, *this review is not meant to subsume the need for a professional security audit of the code*.  Audits of implementation-specific vulnerabilities, improper deployment configuration, etc. are not in scope of a security assessment.  A security assessmet is intended to uncover design and configuration flaws and to obtain a clear, comprehensive articulation of the project's design goals and aspirations while documenting the intended security properties enforced, fulfilled, or executed by said project.
+Due to the nature and timeframe for the analysis, *this review is not meant to
+subsume the need for a professional security audit of the code*.  Audits of
+implementation-specific vulnerabilities, improper deployment configuration, etc.
+are not in scope of a security assessment.  A security assessmet is intended to
+uncover design and configuration flaws and to obtain a clear, comprehensive
```
[the remark this answers is not in the document]
Comment (ultrasaurus) [reply assessments/README.md]: good catch... when we wrote it I think it was intended to describe configurability, but just simplifying to "design flaws" covers it!

**Comment by ultrasaurus** [comment]
I think this is ready to go!

File: assessments/intake-process.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+cloud native ecosystem and helping cloud native projects succeed.
+
+The following process describes how projects are prioritized for security
+assessments.
+
+# Authority
+
+Team members are welcome to submit PRs to streamline process when priorities
+are clear based on the criteria below. As needed, specific leaders (identified
+in the [repo root README](/README.md@#security-assessments)) will coordinate
```
Comment (lumjjb) [inline assessments/intake-process.md]: Broken link because of "@"

Comment (ultrasaurus) [reply assessments/intake-process.md]: fixed!

File: assessments/intake-process.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+# Pre-conditions
+
+* The project is either a CNCF project OR an assertion that the project is cloud
+  native (any objection must be resolved before an assessment would be
+  considered)
+* The project has identified a project lead and has a written self-assessment
+
+# Intake priorities
+
```
Comment (JustinCappos) [inline assessments/intake-process.md]:
```suggestion

Note, this is meant to be informational, not a set of rules that must be followed.  These guidelines are meant to give some
rationale for our thought process, but these are not binding rules.

```

File: assessments/intake-process.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+   year of audit. (For future audits, the security assessment will be a
+   pre-condition to the audit.)
+3. CNCF Projects that request a review (or invited by SIG members), prioritized
+   by project maturity (e.g. graduated projects will be highest priority, then
+   incubated projects, then sandbox).
+4. Non-CNCF Projects that request a review (or invited by SIG members).
+
+# Updates and renewal
+
+The Security Assessment team will aim to review assessed projects annually,
```
Comment (JustinCappos) [inline assessments/intake-process.md]: This somewhat conflicts with 2 above.

Comment (ultrasaurus) [reply assessments/intake-process.md]: I changed this to "aim to" in response to your earlier comment.  does that help?

File: assessments/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (JustinCappos) [reply assessments/README.md]: Okay, let's omit this change if you want.  It's not related to this PR...

File: assessments/intake-process.md (hunk 15 above)
Comment (ultrasaurus) [reply assessments/intake-process.md]: added text to address this to clarify how this would work as guidance -- a lightweight process for us to manage exceptions, keeping TOC in the loop
